# Controller manager configuration
controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    ### Controllers available
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    forward_position_controller:
      type: forward_command_controller/ForwardCommandController
    
    forward_velocity_controller:
      type: velocity_controllers/JointGroupVelocityController

    forward_acceleration_controller: # gazebo Hardware Interface不支持直接加速度控制
      type: forward_command_controller/ForwardCommandController

    joint_effort_controller:
      type: effort_controllers/JointGroupEffortController

    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    ur5_arm_zell_controller:
      type: ur5_arm_zell_controller/RobotController


### Properties of the controllers that we will use and definition of joints to use ###
forward_position_controller:
  ros__parameters:
    joints:
      - ur5_1_shoulder_pan_joint
      - ur5_1_shoulder_lift_joint
      - ur5_1_elbow_joint
      - ur5_1_wrist_1_joint
      - ur5_1_wrist_2_joint
      - ur5_1_wrist_3_joint
    interface_name: position

forward_velocity_controller:
  ros__parameters:
    joints:
      - ur5_1_shoulder_pan_joint
      - ur5_1_shoulder_lift_joint
      - ur5_1_elbow_joint
      - ur5_1_wrist_1_joint
      - ur5_1_wrist_2_joint
      - ur5_1_wrist_3_joint

forward_acceleration_controller:
  ros__parameters:
    joints:
      - ur5_1_shoulder_pan_joint
      - ur5_1_shoulder_lift_joint
      - ur5_1_elbow_joint
      - ur5_1_wrist_1_joint
      - ur5_1_wrist_2_joint
      - ur5_1_wrist_3_joint
    interface_name: acceleration

joint_effort_controller:
  ros__parameters:
    joints:
      - ur5_1_shoulder_pan_joint
      - ur5_1_shoulder_lift_joint
      - ur5_1_elbow_joint
      - ur5_1_wrist_1_joint
      - ur5_1_wrist_2_joint
      - ur5_1_wrist_3_joint
    interface_name: effort

joint_trajectory_controller:
  ros__parameters:
    joints:
      - ur5_1_shoulder_pan_joint
      - ur5_1_shoulder_lift_joint
      - ur5_1_elbow_joint
      - ur5_1_wrist_1_joint
      - ur5_1_wrist_2_joint
      - ur5_1_wrist_3_joint

    command_interfaces:
      - position
      - velocity
      - acceleration

    state_interfaces:
      - position
      - velocity
      - acceleration

    state_publish_rate: 50.0 # Hz, Defaults to 50
    action_monitor_rate: 20.0 # Hz, Defaults to 20
    interpolation_method: splines
    allow_partial_joints_goal: false # Defaults to false
    open_loop_control: true
    allow_integration_in_goal_trajectories: true
    constraints:
      stopped_velocity_tolerance: 0.01 # Defaults to 0.01
      goal_time: 0.0 # Defaults to 0.0 (start immediately)

ur5_arm_zell_controller:
  ros__parameters:
    joints:
      - ur5_1_shoulder_pan_joint
      - ur5_1_shoulder_lift_joint
      - ur5_1_elbow_joint
      - ur5_1_wrist_1_joint
      - ur5_1_wrist_2_joint
      - ur5_1_wrist_3_joint
    interface_name: position
